src/pages/api/logs/[id].js

```diff
@@ -0,0 +1,84 @@
+import mongoose, { Types } from "mongoose";
+import { updateLog } from "../../../../server/db/actions/Log";
+import { z } from "zod";
+import { consts } from "@/utils/consts";
+
+const logSchema = z.object({
```

Comment:
You don't need `.optional()` on each field, if you're using `.partial()` on line 23